src/components/application_manager/src/rpc_service_impl.cc

```diff
@@ -535,13 +527,58 @@ void RPCServiceImpl::SendMessageToMobile(
                (*message)[jhs::S_PARAMS][jhs::S_MESSAGE_TYPE].asInt())) {
     allow_unknown_parameters = false;
   }
+
+  const auto api_function_id = static_cast<mobile_apis::FunctionID::eType>(
+      (*message)[strings::params][strings::function_id].asUInt());
+
+  if (app) {
+    mobile_so_factory().attachSchema(*message, false, app->msg_version());
+    rpc::ValidationReport report("RPC");
+    auto validation_result = message->validate(&report, app->msg_version());
+    LOG4CXX_DEBUG(logger_,
+                  "Attached schema to message, result if valid: "
+                      << validation_result
+                      << "\nError report: " << rpc::PrettyFormat(report));
+
+    if (validation_result != smart_objects::errors::eType::OK) {
```

Comment:
@YarikMamykin please extract it to the separate funciton `SendMessageToMobile` if huge enough

Reply:
@LuxoftAKutsan please check adb85dd4c0c945e43ef981329d5a1ab766c55a48